site/frontend/src/pages/detailed-query/page.vue

```diff
@@ -202,41 +216,82 @@ async function loadData() {
 }
 
 function populateUIData(responseData: SelfProfileResponse, state: Selector) {
-  if (!responseData.base_profile_delta) {
-    showDelta.value = false;
-  }
-
-  if (!state.scenario.includes("incr-")) {
-    showIncr.value = false;
-  }
+  showDelta.value =
+    responseData.base_profile_delta !== undefined &&
+    responseData.base_profile_delta !== null;
+  showIncr.value = state.scenario.includes("incr-");
 }
 
-function sortTable(columnName: string, defaultDirection: number) {
+function changeSortParameters(
+  columnName: string,
+  defaultDirection: SortDirection
+) {
   // Toggle direction if clicking the same column, otherwise use default direction
   if (currentSortColumn.value === columnName) {
     currentSortDirection.value =
       currentSortDirection.value === "asc" ? "desc" : "asc";
   } else {
     currentSortColumn.value = columnName;
-    currentSortDirection.value = defaultDirection === 1 ? "asc" : "desc";
+    currentSortDirection.value = defaultDirection;
   }
 
   // Update URL with new sort state
   storeSortToUrl();
 }
 
-function getSortAttributes(columnName: string) {
-  if (currentSortColumn.value === columnName) {
-    return {
-      "data-sorted-by": currentSortDirection.value,
-    };
+function getHeaderClass(columnName: string): string {
+  if (columnName === currentSortColumn.value) {
+    if (currentSortDirection.value === "asc") {
+      return "header-sort-asc";
+    } else {
+      return "header-sort-desc";
+    }
   }
-  return {};
+  return "header-sort";
 }
 
-onMounted(async () => {
-  await loadData();
-});
+function DeltaComponent({delta}: {delta: DeltaData | null}) {
+  if (delta === null) {
+    return <span>-</span>;
+  }
```

Comment:
this is very unusual pattern in Vue 😄 , does it actually work? Bit of a "Reactism"

Reply:
It requires some build system setup, but it seems to work fine, it's already used elsewhere on the site (`tabs.vue`).